### PR TITLE
Green blood harvested from vali scales off of attack speed.

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
Makes the amount of green blood you gain per swing dependent on your swing speed.
## Why It's Good For The Game
Seeing how claymore has more than double the attack speed as the other Vali weapons (24/11 = 2.18), it really struggles at maintaining green blood.
Compared to spear, you sacrifice 1 tile of range and lose a lot of DPS for a 1-use 90 damage strike.
This brings it more on par with making the green blood gain somewhat more comparable.
## Changelog
:cl:
balance: Vali weapons now gain green blood dependent on their attack speed. Claymore is the most affected by this.
/:cl:

### CODE REVIEW

**Comment by maikilangiolo** [comment]
Do maints even give a single shit about comments on PRs? In the small chance they do, this is a good PR. Double the gain, because you attack twice as slow (yes, you also deal less DPS because it's a single 90 vs two at...54?)

File: code/datums/components/chem_booster.dm [hunk 1]
```diff
@@ -367,15 +367,15 @@
 	return TRUE
 
 ///Handles resource collection and is ativated when attacking with a weapon.
-/datum/component/chem_booster/proc/drain_resource(datum/source, mob/living/M, mob/living/user)
+/datum/component/chem_booster/proc/drain_resource(obj/item/weapon/source, mob/living/M, mob/living/user)
```
Comment (TiviPlus) [inline code/datums/components/chem_booster.dm:370]: incorrect casting

File: code/game/objects/items/weapons/weapon.dm [hunk 2]
```diff
@@ -14,6 +14,7 @@
 	var/max_shots = 0 //codex, energy weapons
 	var/scope_zoom = FALSE//codex
 	var/self_recharge = FALSE //codex
+	var/green_blood_drained = 20
```
Comment (TiviPlus) [inline code/game/objects/items/weapons/weapon.dm:17]: Missing autodoc + if you are changing this due to attack speed why is the resource calculation not just updated to incorporate attackspeed

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/weapon.dm (hunk 2 above)
Comment (Xander3359) [reply code/game/objects/items/weapons/weapon.dm:17]: Sure

File: code/datums/components/chem_booster.dm (hunk 1 above)
Comment (Xander3359) [reply code/datums/components/chem_booster.dm:370]: Isn't the source always a weapon?

**Comment by Xander3359** [comment]
Walance change.

**Pull request closed without merging**